WordPress/gutenberg
# Remove do_shortcode and wpautop for block template parts (still need to remove shortcode block)

Relates to issue #20342

## Description
This PR removes the wpautop and do_shortcode wrappers for block template parts as suggested by @azaozz in #20343.

## How has this been tested?
1. Create a block-based theme, with the header.html and footer.html in the block-template-parts directory, and the index.html file pulling those in. 
2. Attempt to add a shortcode block to the header block-template-part. Notice that you still can (though ultimately this needs to be removed).
3. Save the header and view your site.
4. Notice the shortcode does not get converted.

## Screenshots 

## Types of changes
Bug fix.

## Still to-do: Un-list the shortcode block if editing a block-template or block-template part. 
Un-listing a block in a context does not seem to be possible yet, as blocks are registered without an attached context. For example, I did not see a way to only register the shortcode block if editing the_content. 

<img width="1069" alt="Screen Shot 2020-02-25 at 5 27 26 PM" src="https://user-images.githubusercontent.com/7538525/75293558-6a94e100-57f4-11ea-9e49-1106b9d8b361.png">

The only way I could see to remove the shortcode block was by removing it here:
https://github.com/WordPress/gutenberg/blob/cd851b254347f40f3784f029ccd5937c369907a7/packages/block-library/src/index.js#L117

But that removes it entirely, for all contexts, not just for the block-template-part editor. 

Additionally, since the block-template-part editor is eventually going to be rolled into Full-Site Editing, and editing a `block-template-part` and `the_content` will be possible in that same view (where shortcodes are still allowed) there would have to be a way to disable the shortcode block, but only if editing in a block-template-part area, and otherwise, leave it registered, even if both contexts are being loaded in a single page.


## Checklist:
